src/config.ml

```diff
@@ -0,0 +1,106 @@
+open Base
+open Bot_components.Utils
+
+let toml_of_file file_path = Toml.Parser.(from_filename file_path |> unsafe)
+
+let toml_of_string s = Toml.Parser.(from_string s |> unsafe)
+
+let find k = TomlTypes.Table.find (Toml.key k)
+
+let subkey_value toml_table k k' =
+  TomlLenses.(get toml_table (key k |-- table |-- key k' |-- string))
+
+let list_table_keys toml_table =
+  TomlTypes.Table.fold
+    (fun k _ ks -> TomlTypes.Table.Key.to_string k :: ks)
+    toml_table []
+
+let string_of_mapping =
+  Hashtbl.fold ~init:"" ~f:(fun ~key ~data acc -> acc ^ f "(%s, %s)\n" key data)
+
+let port toml_data =
+  Option.value_map
+    (subkey_value toml_data "server" "port")
+    ~f:Int.of_string
+    ~default:
+      (Option.value_map (Sys.getenv "PORT") ~f:Int.of_string ~default:8000)
+
+let gitlab_access_token toml_data =
+  match subkey_value toml_data "gitlab" "api_token" with
+  | None ->
+      Sys.getenv_exn "GITLAB_ACCESS_TOKEN"
+  | Some secret ->
+      secret
+
+let github_access_token toml_data =
+  match subkey_value toml_data "github" "api_token" with
+  | None ->
+      Sys.getenv_exn "GITHUB_ACCESS_TOKEN"
+  | Some secret ->
+      secret
+
+let github_webhook_secret toml_data =
+  match subkey_value toml_data "github" "webhook_secret" with
+  | None ->
+      Sys.getenv_exn "GITHUB_WEBHOOK_SECRET"
+  | Some secret ->
+      secret
+
+let bot_name toml_data =
+  Option.value_map
+    (subkey_value toml_data "bot" "name")
+    ~f:String.of_string ~default:"coqbot"
```

Comment:
For now, one can configure the bot name and email with environment variable. I'm OK with removing this possibility but let's not forget that the documentation should be updated to reflect that.